dev/io.openliberty.microprofile.lra.1.0.internal/src/io/openliberty/microprofile/lra/internal/LraFilterComponent.java

```diff
@@ -0,0 +1,115 @@
+     * Best practice: this should be a protected method, not public or private
+     *
+     * @param properties : Map containing service & config properties
+     *            populated/provided by config admin
+     */
+    @Activate
+    protected void activate(Map<String, Object> properties) throws LraException {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(tc, "LraFilterComponent activated", properties);
+        }
+        String coordString = "http://" + config.getHost() + ":" + config.getPort() + "/" + config.getPath();
+        if (TraceComponent.isAnyTracingEnabled() && tc.isInfoEnabled()) {
+            Tr.info(tc, "Attempting to contact coordinator at " + coordString);
+        }
+        try {
+            URI coord = new URI(coordString);
+            NarayanaLRAClient.setDefaultCoordinatorEndpoint(coord);
+        } catch (URISyntaxException e) {
+            Tr.error(tc, "Invalid LRA coordinator URI: '" + coordString + "'");
+            throw new LraException(Tr.formatMessage(tc, "LRA_INVALID_COORDINATOR_URI.CWMRX5000E"), e);
+        }
+
+    }
+
+    /**
+     * DS method to deactivate this component.
+     * Best practice: this should be a protected method, not public or private
+     *
+     * @param reason int representation of reason the component is stopping
+     */
+    protected void deactivate(int reason) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(tc, "SampleComponent deactivated, reason=" + reason);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void installProvider(boolean clientSide, List<Object> providers, Set<String> features) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(tc, "Registering LRA filters");
+        }
+
+        if (clientSide) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(tc, "Registering client side filters");
+            }
+            ClientLRARequestFilter requestFilter = new ClientLRARequestFilter();
+            providers.add(requestFilter);
+            ClientLRAResponseFilter responseFilter = new ClientLRAResponseFilter();
+            providers.add(responseFilter);
+        } else {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(tc, "Registering serverside side filters");
+            }
+            try {
+                // Rather unhelpfully, the ServerLRAFilter constructor throws 'Exception'. There isn't much we can do with
+                // that. Re-throwing should prevent the servlet being initialized, which is probably better than swallowing
+                // the exception
+                providers.add(new ServerLRAFilter());
+            } catch (Exception e) {
+                Tr.error(tc, "Exception during registration of LRA filters", e);
```

Comment:
I'm not sure that this error message is required since FFDC will automatically added. If it is needed then it should be NLS enabled.

Reply:
Again, I don't think I'm adding anything useful, so I'll remove it